File: php/features.php
<?php
declare(strict_types=1);

// TheColor SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class TheColorFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TheColorBaseFeature();
            case "test":
                return new TheColorTestFeature();
            default:
                return new TheColorBaseFeature();
        }
    }
}

File: py/thecolor_sdk/features.py
# TheColor SDK feature factory

from thecolor_sdk.feature.base_feature import TheColorBaseFeature
from thecolor_sdk.feature.test_feature import TheColorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TheColorBaseFeature(),
        "test": lambda: TheColorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
